### ISSUE TITLE
Override api code

### ISSUE THREAD

**Issue by ItsRedlyXD** [OrnitheMC/ornithe-standard-libraries#4]
This is rarely done, but it'd be huge for mods like Vulkan mod, almost removing all of it's compatibility issues. It'd be so awesome to have it!!!

**Comment by SRAZKVT** [comment]
Mind explain a bit more around this idea ? One of the goals of OSL is to essentially do as little changes to the actual MC codebase, or at least not any that might have repercussions on gameplay.

**Comment by ItsRedlyXD** [comment]
Yeah, let me explain.

When you have a custom rendering mod (like vulkanmod), you need to override the rendering functions, else almost every mod in existance would crash out.

**Comment by SRAZKVT** [comment]
Yes, and what is your proposal to avoid that ? There isn't really a way to modify a complete pan of the game while being compatible with other mods doing similar things to the same pan of the game.

**Comment by ItsRedlyXD** [comment]
It could maybe be an eventhandler with a default value (sorry, I come from c#), that can be modified.

**Comment by SRAZKVT** [comment]
i don't see how an event would help in this situation

**Comment by SpaceWalkerRS** [comment]
I think this is best done through specific means for each library. So for the Rendering API, OSL should request a custom implementation from loaded mods, and if no implementation is provided, use a default implementation. I don't know if there is any other area where such a system is needed.
